src/ADAL.PCL.CoreCLR/CryptographyHelper.cs

```diff
@@ -25,14 +25,46 @@
 //
 //------------------------------------------------------------------------------
 
+using System;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
 namespace Microsoft.IdentityModel.Clients.ActiveDirectory
 {
     internal class CryptographyHelper : ICryptographyHelper
     {
-
         public string CreateSha256Hash(string input)
         {
-            return null;
+            if (string.IsNullOrWhiteSpace(input))
+            {
+                return null;
+            }
+
+            using (var sha256 = SHA256.Create())
+            {
+                var inputBytes = Encoding.UTF8.GetBytes(input);
+                var outputBytes = sha256.ComputeHash(inputBytes);
+                return Convert.ToBase64String(outputBytes);
+            }
+        }
+
+        public byte[] SignWithCertificate(string message, X509Certificate2 certificate)
+        {
+            if (message == null)
+            {
+                throw new ArgumentNullException(nameof(message));
+            }
+
+            if (certificate == null)
+            {
+                throw new ArgumentNullException(nameof(certificate));
+            }
+
+            var privateKey = certificate.GetRSAPrivateKey();
+            var messageBytes = Encoding.UTF8.GetBytes(message);
+
+            return privateKey.SignData(messageBytes, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
```

Comment:
The reason behind my choice of using the PKCS#1 padding scheme here is to be consistent with [MSAL](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/7fe9410913ff277fe1045e65db8066158720a208/src/Microsoft.Identity.Client/Platforms/netstandard1.3/CryptographyHelper.cs#L68).